src/test/java/org/wso2/extension/siddhi/window/uniquelengthbatch/UniqueLengthBatchWindowTestCase.java

```diff
@@ -0,0 +1,518 @@
+                    }
+                    count++;
+                }
+                Assert.assertEquals("No of emitted events at window expiration", inEventCount - length,
+                        removeEventCount);
+                eventArrived = true;
+            }
+        });
+
+        InputHandler inputHandler = executionPlanRuntime.getInputHandler("cseEventStream");
+        executionPlanRuntime.start();
+        inputHandler.send(new Object[]{"IBM", 700f, 1});
+        inputHandler.send(new Object[]{"WSO2", 60.5f, 2});
+        inputHandler.send(new Object[]{"IBM", 700f, 3});
+        inputHandler.send(new Object[]{"WSO2", 60.5f, 4});
+        inputHandler.send(new Object[]{"IBM", 700f, 5});
+        inputHandler.send(new Object[]{"WSO2", 60.5f, 6});
+        Thread.sleep(500);
+        Assert.assertEquals("In event count", 6, inEventCount);
+        Assert.assertEquals("Remove event count", 4, removeEventCount);
+        Assert.assertTrue(eventArrived);
+        executionPlanRuntime.shutdown();
+    }
+
+    @Test
+    public void LengthBatchWindowTest4() throws InterruptedException {
+
+        SiddhiManager siddhiManager = new SiddhiManager();
+        String cseEventStream = "" +
+                "define stream cseEventStream (symbol string, price float, volume int);";
+        String query = "@info(name = 'query1') " +
+                "from cseEventStream#window.unique:lengthBatch(symbol,4) " +
+                "select symbol,sum(price) as sumPrice,volume " +
+                "insert into outputStream ;";
+
+        ExecutionPlanRuntime executionPlanRuntime = siddhiManager.createExecutionPlanRuntime(cseEventStream + query);
+        executionPlanRuntime.addCallback("outputStream", new StreamCallback() {
+            @Override
+            public void receive(Event[] events) {
+                EventPrinter.print(events);
+                for (Event event : events) {
+//                    Assert.assertEquals("Events cannot be expired", false, event.isExpired());
+                    inEventCount++;
+//                    if (inEventCount == 1) {
+//                        Assert.assertEquals(130.0, event.getData(1));
+//                    }
+                }
+                eventArrived = true;
+            }
+        });
+
+        InputHandler inputHandler = executionPlanRuntime.getInputHandler("cseEventStream");
+        executionPlanRuntime.start();
+        inputHandler.send(new Object[]{"IBM", 10f, 0});
+        inputHandler.send(new Object[]{"IBM", 20f, 1});
+        inputHandler.send(new Object[]{"IBM", 30f, 0});
+        inputHandler.send(new Object[]{"IBM", 40f, 1});
+        inputHandler.send(new Object[]{"IBM", 50f, 0});
+        inputHandler.send(new Object[]{"IBM", 60f, 1});
+        Thread.sleep(500);
+        executionPlanRuntime.shutdown();
+//        Assert.assertEquals(0, inEventCount);
+//        Assert.assertTrue(eventArrived);
```

Comment:
@hmrajas : Is there any particular reason to comment out the above assertions? Same applies to line 263,265-267.